### PR TITLE
osdc: standardize osdc code with std::atomic

### PR DESCRIPTION
**Pull request by cxwshawn**
standalize ceph code with std::atomic other than ceph::atomic
same work as https://github.com/ceph/ceph/pull/9138
Signed-off-by: Xiaowei Chen chen.xiaowei@h3c.com

### CODE REVIEW

**Comment by liewegas** [comment]
Can you squash these into 1 commit so they both build?

**Comment by cxwshawn** [comment]
@liewegas sorry, have squashed, please review.

File: src/osdc/Objecter.h [hunk 1]
```diff
@@ -1144,16 +1145,16 @@ class Objecter : public md_config_obs_t, public Dispatcher {
   using Dispatcher::cct;
   std::multimap<string,string> crush_location;
 
-  atomic_t initialized;
+  atomic_bool initialized;
```
Comment (cbodley) [inline src/osdc/Objecter.h]: similar to https://github.com/ceph/ceph/pull/9138#discussion_r64237097, can we qualify these with `std::`?

**Comment by cxwshawn** [comment]
@cbodley thanks for your advice, @liewegas  fixed, please review.

**Comment by cxwshawn** [comment]
@liewegas @tchaikov PING!

**Pull request closed without merging**